system-test/bigtable.js

```diff
@@ -0,0 +1,289 @@
+        done();
+      });
+    });
+
+  });
+
+  describe('column families', function() {
+    var FAMILY_NAME = 'presidents';
+    var FAMILY = TABLE.family(FAMILY_NAME);
+
+    before(function(done) {
+      FAMILY.create(done);
+    });
+
+    it('should get a list of families', function(done) {
+      TABLE.getFamilies(function(err, families) {
+        assert.ifError(err);
+        assert.strictEqual(families.length, 1);
+        assert.strictEqual(families[0].name, FAMILY.name);
+        done();
+      });
+    });
+
+    it('should get a family', function(done) {
+      var family = TABLE.family(FAMILY_NAME);
+
+      family.get(function(err) {
+        assert.ifError(err);
+        assert.strictEqual(family.name, FAMILY.name);
+        done();
+      });
+    });
+
+    it('should check if a family exists', function(done) {
+      FAMILY.exists(function(err, exists) {
+        assert.ifError(err);
+        assert.strictEqual(exists, true);
+        done();
+      });
+    });
+
+    it('should check if a family does not exist', function(done) {
+      var family = TABLE.family('prezzies');
+
+      family.exists(function(err, exists) {
+        assert.ifError(err);
+        assert.strictEqual(exists, false);
+        done();
+      });
+    });
+
+    it('should get the column family metadata', function(done) {
+      FAMILY.getMetadata(function(err, metadata) {
+        assert.ifError(err);
+        assert(FAMILY.metadata, metadata);
+        done();
+      });
+    });
+
+    it('should update a column family', function(done) {
+      var maxAge = {
+        seconds: 10000,
+        nanos: 10000
+      };
+
+      var metadata = {
+        maxAge: maxAge
+      };
+
+      FAMILY.setMetadata(metadata, function(err, metadata_) {
+        assert.ifError(err);
+
+        var maxAge_ = metadata_.gcRule.maxAge;
+        // for some reason this comes back as a string
+        assert.equal(maxAge_.seconds, maxAge.seconds);
+        assert.strictEqual(maxAge_.nanas, maxAge.nanas);
+        done();
+      });
+    });
+
+    it('should delete a column family', function(done) {
+      FAMILY.delete(function(err) {
+        assert.ifError(err);
+        done();
+      });
+    });
+
+  });
+
+  describe('rows', function() {
+
+    before(function(done) {
+      // consider a TABLE.createFamilies() method?
+      async.each(['name', 'life'], function(family, callback) {
+        TABLE.createFamily(family, callback);
+      }, done);
+    });
+
+    it('should insert a row', function(done) {
+      var data = {
+        name: {
+          first: 'Abraham',
+          last: 'Lincoln'
+        },
+        life: {
+          born: 'Feb. 12, 1809',
+          died: 'April 15, 1865'
+        }
+      };
+
+      TABLE.createRow('lincoln', data, function(err) {
+        assert.ifError(err);
+        done();
+      });
+    });
+
+    it('should insert multiple rows', function(done) {
+      var rows = {
```

Comment:
Probably the datastore style. It can be awkward to assign a var as a key to an object, i.e.

```js
var key = 'my-key-reference'

var row = {}
row[key] = {..}
// vs
var row = {
  key: key,
  data: {..}
}
```

And this might be premature (haven't given a fair review to the entire API yet), but what about `table.insert` instead of `createRow` and `createRows`?